### PR TITLE
Chores/best practices dah 729

### PR DESCRIPTION
**Pull request by software-project**
Adds linting and IDE best practices with prettier: https://sfgovdt.jira.com/browse/DAH-729

This branch depends on PR: https://github.com/SFDigitalServices/sf-dahlia-web/pull/1438

### CODE REVIEW

**Comment by software-project** [comment]
@james-wills-sf `yarn unit` is failing on this one and I can't figure out why. It was passing on the circle ci test branch that this one is branched out of.

File: .eslintrc.js [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    "lighthouserc.js",
+    "public",
+    "node_modules",
+    "spec/e2e",
+    "spec/javascripts",
+    "**/.eslintrc.js"
+  ],
+  settings: {
+    react: {
+      // Must be updated when package.json react version is bumped
```
Comment (james-wills-sf) [inline .eslintrc.js:110]: Can you add a line to the package.json react definition pointing to this file as well? I think it's likely we'd forget to update both if there isn't a comment

Comment (james-wills-sf) [reply .eslintrc.js:110]: Edit: nevermind since I guess you can't add comments to a .json file.

**Review by james-wills-sf — COMMENTED** [review]
I left some comments with a few adjustments to the rules and setup, but can you also add a pre-commit hook to check for any linting or formatting errors and block on errors?

We also want a CircleCI check to make sure you can't merge with linting errors

**Review by james-wills-sf — APPROVED** [review]
Changes lgtm

**Pull request merged**